### PR TITLE
kernel crate: Improve comments and generated docs

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This is a pass on some of the kernel crate to improve the comments and generated documentation. Mostly this is to try to further leon-ify our docs. There's also some wrapping in there too.

### Testing Strategy

`cargo doc`

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by lschuermann — APPROVED** [review]

File: kernel/src/syscall.rs [hunk 1]
```diff
@@ -4,10 +4,67 @@
 
 //! Mechanisms for handling and defining system calls.
 //!
-//! This includes:
-//! - syscall class
-//! - error types
-//! - interface trait for context switches
+//! # System Call Overview
```
Comment (ppannuto) [inline kernel/src/syscall.rs:7]: Is this duplicating documentation we have elsewhere and we would be possibly better off just linking to?

(I recognize that this is just moving an existing comment, but I'm wondering if something so lengthy, and more general in description is better suited to the book or similar)

**Review by ppannuto — APPROVED** [review]
Man, what meeting did you not want to be in today? :)

File: kernel/src/syscall.rs (hunk 1 above)
Comment (bradjc) [reply kernel/src/syscall.rs:7]: I think it is duplicated, but it helps provide context for the file.

Comment (alevy) [reply kernel/src/syscall.rs:7]: I agree with @bradjc, I prefer this to be in inline docs _as well_

**Pull request merged**